torch/_meta_registrations.py

```diff
@@ -7294,37 +7320,134 @@ def grouped_mm(
     out_dtype = out_dtype or mat1.dtype
     torch._check(bias is None, lambda: "bias not supported yet")
 
-    def _compute_grouped_gemm_output_size(mat1, mat2, offs):
-        mat1_is_2d = mat1.dim() == 2
-        mat2_is_2d = mat2.dim() == 2
-
-        if mat1_is_2d:
-            if mat2_is_2d:
-                return offs.size(0), mat1.size(0), mat2.size(1)
-            else:
-                torch._check(
-                    offs.size(0) == mat2.size(0), "matrix batch sizes have to match"
-                )
-                return mat1.size(0), mat2.size(-1)
-        else:
-            if mat2_is_2d:
-                torch._check(
-                    offs.size(0) == mat1.size(0), "matrix batch sizes have to match"
-                )
-                return mat1.size(1), mat2.size(1)
-            else:
-                # regular bmm
-                torch._check(
-                    mat1.size(0) == mat2.size(0), "batched dimension has to match"
-                )
-                return mat1.size(0), mat1.size(1), mat2.size(-1)
-
     out_size = _compute_grouped_gemm_output_size(mat1, mat2, offs)
     out = mat1.new_empty(out_size, dtype=out_dtype)
 
     return out
 
 
+@register_meta([aten._scaled_grouped_mm.default])
+def meta_scaled_grouped_mm(
+    mat_a: torch.Tensor,
+    mat_b: torch.Tensor,
+    scale_a: torch.Tensor,
+    scale_b: torch.Tensor,
+    offs: Optional[torch.Tensor] = None,
+    bias: Optional[torch.Tensor] = None,
+    scale_result: Optional[torch.Tensor] = None,
+    out_dtype: Optional[torch.dtype] = None,
+    use_fast_accum: bool = False,
+):
+   # Check dimensions
+    torch._check(mat_a.dim() == 2 or mat_a.dim() == 3, lambda: "mat_a has to be 2 or 3d")
+    torch._check(mat_b.dim() == 2 or mat_b.dim() == 3, lambda: "mat_b has to be 2 or 3d")
+
+    a_is_2d = mat_a.dim() == 2
+    b_is_2d = mat_b.dim() == 2
+
+    # Check offsets
+    torch._check((offs is not None) == (a_is_2d or b_is_2d), lambda: "Have to provide offsets if there is a 2d matrix")
+
+    if offs is not None:
+        torch._check(offs.dim() == 1, lambda: "offs has to be 1D")
+        torch._check(offs.dtype == torch.int, lambda: "Offsets have to be int32")
+
+    # Check matrix sizes
+    torch._check(
```

Comment:
Should some of these be torch._check_value or torch._check_assert?

Reply:
I'm using the same macros as are used in _scaled_mm meta, none of them are check_assert.